pulp-glue/pulp_glue/common/authentication.py

```diff
@@ -0,0 +1,94 @@
+import json
+import typing as t
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import click
+import requests
+
+
+class OAuth2ClientCredentialsAuth(requests.auth.AuthBase):
+    def __init__(
+        self,
+        client_id: str = "",
+        client_secret: str = "",
+        token_url: str = "",
+        scopes: t.List[str] = [],
+        *args: t.List[t.Any],
+        **kwargs: t.Dict[t.Any, t.Any],
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_url = token_url
+        self.scopes = scopes
+        self.token: t.Dict[t.Any, t.Any] = {}
+
+    def __call__(self, request: requests.PreparedRequest) -> requests.PreparedRequest:
+        self.retrieve_local_token()
+
+        access_token = self.token.get("access_token")
+        request.headers["Authorization"] = f"Bearer {access_token}"
+
+        request.register_hook("response", self.handle401)  # type: ignore
+
+        return request
+
+    def handle401(
+        self, response: requests.Response, **kwargs: t.Dict[t.Any, t.Any]
```

Comment:
This is not how you annotate kwargs:
```suggestion
        self, response: requests.Response, **kwargs: t.Any
```